store/src/java/com/zimbra/cs/service/util/S3BucketGlobalConfigUtil.java

```diff
@@ -0,0 +1,485 @@
+    
+    /**
+     * Parse the Attr list inside of request element to a GlobalS3BucketConfiguration object
+     *
+     * @param request
+     * @return GlobalS3BucketConfiguration
+     * @throws ServiceException
+     */
+    public static GlobalS3BucketConfiguration parseCreateRequestToGlobalS3BucketConfig(CreateS3BucketConfigRequest request) throws ServiceException {
+    	
+    	GlobalS3BucketConfiguration gs3bc = new GlobalS3BucketConfiguration();
+    		
+		for (Attr attr : request.getAttrs()) {
+			setValueByReflection(gs3bc, attr.getKey(), attr.getValue());
+		}
+		
+		return gs3bc;
+    	
+    }
+    
+    /**
+     * Parse the Attr list inside of request element to a GlobalS3BucketConfiguration object
+     *
+     * @param request
+     * @return GlobalS3BucketConfiguration
+     * @throws ServiceException
+     */
+    public static GlobalS3BucketConfiguration parseDeleteRequestToGlobalS3BucketConfig(DeleteS3BucketConfigRequest request) throws ServiceException {
+    	
+    	GlobalS3BucketConfiguration gs3bc = new GlobalS3BucketConfiguration();
+    		
+		for (Attr attr : request.getAttrs()) {
+			setValueByReflection(gs3bc, attr.getKey(), attr.getValue());
+		}
+		
+		return gs3bc;
+    	
+    }
+    
+    /**
+     * Given an object, set the value for a specific field using Java Reflection API.
+     *
+     * @param object
+     * @param fieldStr
+     * @param valueStr
+     * @return
+     * @throws ServiceException
+     */
+    private static void setValueByReflection(Object object, String fieldStr, String valueStr) throws ServiceException {
+		
+		boolean breakBucle = false;
+		
+		try {
+			
+			if(object != null && fieldStr != null && !fieldStr.isEmpty()) {
+				Class<? extends Object> clazz = object.getClass();
+				for (Field field : clazz.getDeclaredFields()) {
+					if(field.getName().equals(fieldStr)) {
+						for (Method method : clazz.getMethods()) {
+							if((method.getName().startsWith("set")) 
+									&& (method.getName().length() == field.getName().length() + 3)
+									&& (method.getName().toLowerCase().endsWith(field.getName().toLowerCase()))) {
+									
+								method.invoke(object, valueStr);
+								breakBucle = true;
+
+							}
+							if(breakBucle)	break;
+						}
+					}
+					if(breakBucle)	break;
+				}
+			}
+			
+		} catch (Exception e) {
+			ZimbraLog.mailbox.warn("Failed to execute set method by reflection", e);
+			throw ServiceException.FAILURE("Failed to execute set method by reflection", null);
+		}
+		
+	}
+
+    /**
+     * Validate the request parameters before to add a bucket
+     *
+     * @param request
+     * @return
+     * @throws ServiceException
+     */
+	public static boolean validateReqBeforeCreate(CreateS3BucketConfigRequest request) throws ServiceException {
+		
+		GlobalS3BucketConfiguration gs3bc = parseCreateRequestToGlobalS3BucketConfig(request);
+		
+		// General validations
+		
+		if(gs3bc.getStoreProvider() == null || gs3bc.getStoreProvider().isEmpty()) {
```

Comment:
add space after if here and below every where

Reply:
done!